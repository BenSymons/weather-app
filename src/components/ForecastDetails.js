import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, humidity, temperature, wind } = forecast;
  const options = { weekday: "long", month: "long", day: "numeric" };
  const formattedDate = new Date(date).toLocaleDateString("en-GB", options);

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        <h2>{formattedDate}</h2>
      </div>
      <div className="forecast-details__max-temperature">
        High: {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__min-temperature">
        Low: {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
      <div className="forecast-details__wind">
        Wind: {wind.speed}mph {wind.direction}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    humidity: PropTypes.number,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
