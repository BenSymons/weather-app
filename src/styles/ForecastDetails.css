
.forecast-details > * {
padding-block-start: 10px;
font-weight: 600;
text-align: center;
font-size: 1.5rem;
}