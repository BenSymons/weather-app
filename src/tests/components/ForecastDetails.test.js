import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1525046400000,
    humidity: 80,
    wind: {
      speed: 30,
      direction: "s",
    },
    temperature: {
      max: 12,
      min: 12,
    },
  };
  it("renders forecast details", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders correct values for props", () => {
    const { getByTestId, getByText } = render(
      <ForecastDetails forecast={validProps} />
    );

    expect(getByText("Monday, 30 April")).toHaveAttribute(
      "class",
      "forecast-details__date"
    );
    expect(getByText("High: 12°C")).toHaveAttribute(
      "class",
      "forecast-details__max-temperature"
    );
    expect(getByText("Low: 12°C")).toHaveAttribute(
      "class",
      "forecast-details__min-temperature"
    );
    expect(getByText("Humidity: 80%")).toHaveAttribute(
      "class",
      "forecast-details__humidity"
    );
    expect(getByText("Wind: 30mph s")).toHaveAttribute(
      "class",
      "forecast-details__wind"
    );
    expect(getByTestId("temperature-div")).toContainElement(
      getByTestId("temperature-div__max")
    );
    expect(getByTestId("temperature-div")).toContainElement(
      getByTestId("temperature-div__min")
    );
  });
});
