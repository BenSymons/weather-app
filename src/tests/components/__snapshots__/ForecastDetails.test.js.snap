// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders forecast details 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      <h2
        class="forecast-details__date"
      >
        Monday, 30 April
      </h2>
    </div>
    <div
      class="forecast-details__temperature"
      data-testid="temperature-div"
    >
      <div
        class="forecast-details__max-temperature"
        data-testid="temperature-div__max"
      >
        High: 12°C
      </div>
      <div
        class="forecast-details__min-temperature"
        data-testid="temperature-div__min"
      >
        Low: 12°C
      </div>
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 80%
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind: 30mph s
    </div>
  </div>
</DocumentFragment>
`;
